newrelic/structures_newrelic_nrql_alert_condition.go

```diff
@@ -304,6 +326,41 @@ func expandNrqlTerms(d *schema.ResourceData, conditionType string) ([]alerts.Nrq
 	return expandedTerms, nil
 }
 
+// NerdGraph
+func expandExpiration(d *schema.ResourceData) (*alerts.AlertsNrqlConditionExpiration, error) {
+	var expiration alerts.AlertsNrqlConditionExpiration
+
+	expiration.OpenViolationOnExpiration = d.Get("open_violation_on_expiration").(bool)
+	expiration.CloseViolationsOnExpiration = d.Get("close_violations_on_expiration").(bool)
+
+	// 0 is not a valid expiration duration so don't set it if it's nonexistent
+	if expirationDuration, ok := d.GetOk("expiration_duration"); ok {
+		v := expirationDuration.(int)
+		expiration.ExpirationDuration = &v
+	}
+
+	return &expiration, nil
+}
+
+// NerdGraph
+func expandSignal(d *schema.ResourceData) (*alerts.AlertsNrqlConditionSignal, error) {
+	var signal alerts.AlertsNrqlConditionSignal
+
+	mappedFillOption := fillOptionMap[d.Get("fill_option").(string)]
```

Comment:
We will likely need some case transformation in the expand and flatten functions for the new enum types. Right now if a user tries to set `fill_option` to `Last_Value` or `last_valuE`, etc, this will probably break. For UX purposes, we try to keep things case-insensitive to provide slightly easier UX for users, but that means a little extra work for us in the code to avoid state drift and errors 😎 

There are a few ways to accomplish this. One pattern is to always store the value as lowercase in the state. Take a look at [`threshold_occurrences`](https://github.com/newrelic/terraform-provider-newrelic/blob/523bbac5cfb78ac1ba5a6c7899726f5efa84e5b2/newrelic/resource_newrelic_nrql_alert_condition.go#L58-L66) and how it's handled in the expand and flatten functions. This won't necessarily be the exact same pattern for `fill_option`, but it will be probably be similar. And if you have other ideas on how to accomplish this, we're all about feedback and ideas! 🙂

Reply:
Good catch. I will ensure these are lowercased in the flatten/expand functions.